airflow/kubernetes/kubernetes_helper_functions.py

```diff
@@ -26,22 +26,7 @@
 log = logging.getLogger(__name__)
 
 
-def _strip_unsafe_kubernetes_special_chars(string: str) -> str:
-    """
-    Kubernetes only supports lowercase alphanumeric characters, "-" and "." in
-    the pod name.
-    However, there are special rules about how "-" and "." can be used so let's
```

Comment:
Mostly for my own curiosity/learning: What are the special rules? Do they no longer cause us issues? Did they ever cause us issues?

Reply:
A pod name must start and end with an alphanumeric character, i.e. `[a-z0-9]`.   So you can't have a leading or trailing `-` or `.`.

If you try to create a pod that violates this, you'll get this error:
```
a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')
```